src/Microsoft.AspNet.Mvc.Core/DefaultControllerFactory.cs

```diff
@@ -38,8 +38,19 @@ public DefaultControllerFactory(IControllerActivator controllerActivator)
             _getPropertiesToActivate = GetPropertiesToActivate;
         }
 
+        /// <summary>
+        /// The <see cref="IControllerActivator"/> used to create a controller.
+        /// </summary>
+        protected IControllerActivator ControllerActivator
```

Comment:
I don't get the arguments why this is bad. Transient is the 'caller-decides' policy. We're increasing the lifetime of the activator passed to this class in the constructor by DI - but not any other instances, let's say requested by user code.

If the activator **owns** a cache I think there's an argument to be made for making it a singleton.

TLDR: Picked most flexible lifetime, used it in the way we want.